src/index.ts

```diff
@@ -3,32 +3,32 @@ let menorValor: number = 18;
 let mayorValor: number = 40;
 let menores21: number = 0;
 let mayores21: number = 0;
-let edadesPermitidas: number = 0;
+let cantidadPersonas: number = 270;
+let edadesPersonas: number[] = new Array(cantidadPersonas);
 
 //funcion que me va a retornar valores aleatorios del 1 al 40, gracias a math random,
 //y math floor redondea
 function edadAleatoria(menorValor: number, mayorValor: number): number {
   let numRandom: number = 0;
-  numRandom = Math.floor(Math.random() * 41);
-  //verifico los valores
-  console.log(numRandom);
-  //tomo el valor si se cumplen las dos a la vez, en caso contrario es menor que 18
-  if (menorValor <= numRandom && mayorValor >= numRandom) {
-    edadesPermitidas = numRandom;
-  }
-  //verifico el valor tomado
-  console.log("edad aleatoria tomada:", edadesPermitidas);
-  return edadesPermitidas;
+  numRandom = Math.floor(
+    Math.random() * (mayorValor - menorValor + 1) + menorValor
+  );
+  return numRandom;
 }
 
 //No especifica si se tiene que acionar de alguna manera el programa pero coloco todo dentro del for
 //puedo guardar las edades en dos Array y despues por medio de otro for puedo saber las cantidades exactas por edad, pero tampoco especifica
 //repito el for 270 veces auto accionando la funcion edadAleatoria
-for (let control = 0; control < 270; control++) {
-  edadesPermitidas = edadAleatoria(menorValor, mayorValor);
-  if (edadesPermitidas > 21) {
+for (let index = 0; index < cantidadPersonas; index++) {
+  edadesPersonas[index] = edadAleatoria(menorValor, mayorValor);
+}
+
+for (let index = 0; index < cantidadPersonas; index++) {
+  if (edadesPersonas[index] > 21) {
     mayores21 += 1;
-  } else menores21 += 1;
+  } else {
+    menores21 += 1;
+  }
 }
```

Comment:
- Agregamos un **for** que se usa para determinar el arreglo **edadesPersonas**, para esto en cada vuelta del for se llama a la función **edadAleatoria** que retorna un valor aleatorio válido y ese valor lo vamos guardando en cada posición del arreglo.